provisioning/miniprov/hare_mp/utils.py

```diff
@@ -114,17 +114,18 @@ def import_kv(self, conf_dir_path: str):
                                str(item_data['value']))
 
 
-def execute(cmd: List[str], env=None) -> str:
+def execute(cmd: List[str], encoding=None, env=None, stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE, inp=None) -> str:
     process = subprocess.Popen(cmd,
-                               stdin=subprocess.PIPE,
-                               stdout=subprocess.PIPE,
+                               stdin=stdin,
```

Comment:
Just for your information: perhaps the `Program` API from helper module may be less wordy and more convenient. 

Example how it is used in `configure` script:
https://github.com/Seagate/cortx-hare/blob/220d3202f74bb32be6479956d3cbce4b3c28a52e/hax/helper/configure.py#L70-L81

Reply:
Moreover, it turns out that if you're trying to use subprocess to run piped commands (when one program's output becomes an input of the next one) is a [bit tricky](https://docs.python.org/3/library/subprocess.html#replacing-shell-pipeline): you need to forcibly close the output stream of the previous command. That is already made within Program/Executor API.